web/libs/datamanager/src/utils/urlJSON.js

```diff
@@ -0,0 +1,20 @@
+// Library that serializes and deserializes JSON data to/from URL-safe strings
+// Uses gzip to minimize the size of the serialized data
+
+import { gzip, ungzip } from 'pako'
+import { fromUint8Array, toUint8Array } from 'js-base64'
+import { packJSON } from './packJSON';
+
+export const deserializeJsonFromUrl = (s) => {
+    // Fall back to packJSON if deserialization fails; this probably means
+    // we are deserializing an old virtual view. TODO(jo): deprecate packJSON entirely
+    try {
+        return JSON.parse(ungzip(toUint8Array(s), { to: 'string' }));
+    } catch (e) {
+        console.log('Error deserializing gzipped data:', e);
```

Comment:
Will the managed exception provide us a reference the the old virtual view that is causing the error?

Reply:
not currently - if we get here, all we know is that we couldn't deserialize with gzip, so we try packJSON instead; for properly encoded packJSON query params (ie, no datetimes because those are broken), the user will get their intended virtual tab seamlessly. If the query string is neither valid gzip nor valid packjson, the packjson call fails silently and we get an empty virtual tab (which is actually not a terrible experience)